src/site/content/en/blog/streams/index.md

```diff
@@ -0,0 +1,966 @@
+which causes any future interactions with the associated stream to error.
+
+#### The `queuingStrategy`
+
+The second, likewise optional, argument of the `WritableStream()` constructor is `queuingStrategy`.
+It is an object that optionally defines a queuing strategy for the stream, which takes two
+parameters:
+
+- `highWaterMark`: A non-negative integer that defines the total number of chunks that can be
+  contained in the internal queue before backpressure is applied.
+- `size(chunk)`: A method containing a parameter chunk that indicates the size to use for each
+  chunk, in bytes.
+
+{% Aside %} You could define your own custom `queuingStrategy`, or use an instance of
+[`ByteLengthQueuingStrategy`](https://developer.mozilla.org/en-US/docs/Web/API/ByteLengthQueuingStrategy)
+or [`CountQueuingStrategy`](https://developer.mozilla.org/en-US/docs/Web/API/CountQueuingStrategy)
+for this object value. If no `queuingStrategy` is supplied, the default used is the same as a
+`CountQueuingStrategy` with a `highWaterMark` of `1`. {% endAside %}
+
+#### The `getWriter()` and `write()` methods
+
+In order to write to a writable stream, you need a writer, which will be a
+`WritableStreamDefaultWriter`. The `getWriter()` method of the `WritableStream` interface returns a
+new instance of `WritableStreamDefaultWriter` and locks the stream to that instance. While the
+stream is locked, no other writer can be acquired until this one is released.
+
+The [`write()`](https://developer.mozilla.org/en-US/docs/Web/API/WritableStreamDefaultWriter/write)
+method of the
+[`WritableStreamDefaultWriter`](https://developer.mozilla.org/en-US/docs/Web/API/WritableStreamDefaultWriter)
+interface writes a passed chunk of data to a `WritableStream` and its underlying sink, then returns
+a promise that resolves to indicate the success or failure of the write operation. Note that what
+"success" means is up to the underlying sink; it might indicate that the chunk has been accepted,
+and not necessarily that it is safely saved to its ultimate destination.
+
+#### The `locked` property
+
+You can check if a writable stream is locked by accessing its
+[`WritableStream.locked`](https://developer.mozilla.org/en-US/docs/Web/API/WritableStream/locked)
+property.
+
+### Writable stream code sample
+
+The code sample below shows all steps in action.
+
+```js
+const writableStream = new WritableStream({
+  async start(controller) {
+    console.log('[start]');
+  },
+  async write(chunk, controller) {
+    console.log('[write]', chunk);
+    // Wait for next write.
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  },
+  async close(controller) {
+    console.log('[close]');
+  },
+  async abort(reason) {
+    console.log('[abort]', reason);
+  },
+});
+
+(async () => {
+  const writer = writableStream.getWriter();
+  const start = Date.now();
+  for (const char of 'abcd') {
+    // Wait to add write queue.
+    await writer.ready;
```

Comment:
Was this example to show the use of `ready` or could it have been shorter with something like below?
```javascript
  for (const char of 'abcd') {
    await writer.write(char);
  }
  await writer.close();
```

Reply:
The idea was to show `.ready` indeed.